### PR TITLE
delete(old demux post processing workflow)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

This PR aims to delete the old demux post processing workflow logic as the new workflow was implemented. Part of https://github.com/Clinical-Genomics/project-planning/issues/482

We need to have a refinement meeting on this PR to double check the functionality we are deleting here is fully covered or not necessary anymore. @henrikstranneheim, @karlnyr 

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/cli/demultiplex/add.py [hunk 1]
```diff
@@ -1,70 +0,0 @@
...
-    report_content: List[str] = post_processing_api.get_report_data(
-        flow_cell_id=flow_cell_id, project_name=project
-    )
-
-    click.echo("\t".join(STATS_HEADER))
-    if not report_content:
-        LOG.warning(f"Could not find any samples for flow cell {flow_cell_id}, project {project}")
-        return
-    for report_line in report_content:
-        click.echo(report_line)
```
Comment (ChrOertlin) [inline cg/cli/demultiplex/add.py:70]: do we require a new implementation to fetch stats for samples related to a ticket?

File: cg/cli/demultiplex/add.py [hunk 2]
```diff
@@ -1,70 +0,0 @@
...
-from cg.models.demultiplex.flow_cell import FlowCellDirectoryData
-
-LOG = logging.getLogger(__name__)
-
-
-@click.command(name="add")
-@OPTION_BCL_CONVERTER
-@click.argument("flow-cell-name")
-@click.pass_obj
-def add_flow_cell_cmd(context: CGConfig, flow_cell_name: str, bcl_converter: str):
```
Comment (ChrOertlin) [inline cg/cli/demultiplex/add.py:29]: this become obsolete as we deprecate cgstats

File: cg/cli/demultiplex/finish.py [hunk 3]
```diff
@@ -21,80 +18,27 @@ def finish_group():
...
         raise click.Abort
 
 
 @finish_group.command(name="flow-cell")
 @click.argument("flow-cell-name")
-@OPTION_BCL_CONVERTER
 @DRY_RUN
 @click.option("--force", is_flag=True)
 @click.pass_obj
-def finish_flow_cell(
```
Comment (ChrOertlin) [inline cg/cli/demultiplex/finish.py:49]: we do not currently have a --force flag for the `finish_flow_cell_temporary` I will make a PR for the implementation of it

File: cg/cli/demultiplex/finish.py [hunk 4]
```diff
@@ -21,80 +18,27 @@ def finish_group():
...
-
-
-@finish_group.command(name="temporary-all")
-@click.pass_obj
-@DRY_RUN
-def finish_flow_cell_temporary_all(context: CGConfig, dry_run: bool):
-    # Temporary finish flow cell logic will replace logic above when validated
-    demux_post_processing_api_temp: DemuxPostProcessingAPI = DemuxPostProcessingAPI(config=context)
-    demux_post_processing_api_temp.set_dry_run(dry_run=dry_run)
-    demux_post_processing_api_temp.finish_all_flow_cells_temp()
```
Comment (ChrOertlin) [inline cg/cli/demultiplex/finish.py:87]: these are the temp command strictly used for testing the new workflow designed to be removed

File: cg/cli/demultiplex/finish.py [hunk 5]
```diff
@@ -21,80 +18,27 @@ def finish_group():
...
-@DRY_RUN
-@click.pass_obj
-def finish_all_hiseq_x(context: CGConfig, dry_run: bool) -> None:
-    """Command to post-process new demultiplexed Hiseq X flow cells."""
-    logging.debug("Checking for new Hiseq X demultiplexed flow cells")
-    demux_post_processing_api: DemuxPostProcessingHiseqXAPI = DemuxPostProcessingHiseqXAPI(
-        config=context
-    )
-    demux_post_processing_api.set_dry_run(dry_run=dry_run)
-    demux_post_processing_api.finish_all_flow_cells(bcl_converter=BclConverter.BCL2FASTQ.value)
```
Comment (ChrOertlin) [inline cg/cli/demultiplex/finish.py:100]: the new demux post processing workflow incorporate novaseq6000, novaseqX and hiseq flow cell demux post processing

File: cg/meta/demultiplex/demux_post_processing.py [hunk 6]
```diff
@@ -156,300 +142,3 @@ def store_flow_cell_data(self, parsed_flow_cell: FlowCellDirectoryData) -> None:
...
-            LOG.info("Files have already been renamed")
-        else:
-            # not needed
-            self.rename_files(demux_results=demux_results)
-        # depcrecated
-        self.add_to_cgstats(demux_results=demux_results)
-        # deprecated
-        self.create_cgstats_reports(demux_results=demux_results)
-        if demux_results.bcl_converter == "bcl2fastq":
-            # relevant - topUnknownbarcodes
```
Comment (ChrOertlin) [inline cg/meta/demultiplex/demux_post_processing.py:404]: I suppose this needs refinement

File: cg/cli/demultiplex/finish.py (hunk 3 above)
Comment (ChrOertlin) [reply cg/cli/demultiplex/finish.py:49]: https://github.com/Clinical-Genomics/cg/pull/2288

File: cg/meta/demultiplex/demux_post_processing.py [hunk 7]
```diff
@@ -156,300 +142,3 @@ def store_flow_cell_data(self, parsed_flow_cell: FlowCellDirectoryData) -> None:
...
-                flow_cell_path=flow_cell_path, bcl_converter=bcl_converter
-            )
-        except FlowCellError:
-            return
-        demux_results: DemuxResults = DemuxResults(
-            demux_dir=Path(self.demux_api.out_dir, flow_cell_name),
-            flow_cell=flow_cell,
-            bcl_converter=bcl_converter,
-        )
-        if not demux_results.flow_cell.is_hiseq_x_copy_completed():
```
Comment (seallard) [inline cg/meta/demultiplex/demux_post_processing.py:261]: We can also remove a lot of functionality from the `FlowCellDirectoryData` class, such as this `is_hiseq_x_copy_completed`

Comment (seallard) [reply cg/meta/demultiplex/demux_post_processing.py:261]: And the associated tests

**Review by seallard — COMMENTED** [review]
These functions and their associated tests can also be removed (since they are only referenced in the tests)
- `create_novaseq_flowcell` in `cg/apps/cgstats/crud/create.py` (the entire dir should be deleted, but this one is not referenced anywhere now in this PR)
- `transfer_flow_cell` in `demux_post_processing.py`
- `rename_project_directory` in `cg/meta/demultiplex/files.py`
- `raw_projects` property in `cg/models/demultiplex/demux_results.py`
- `BclConvertSampleSheetData` in `cg/apps/sequencing_metrics_parser/models/bcl_convert.py`
- `files_renamed` in `cg/models/demultiplex/demux_results.py`
- `demux_sample_sheet_path` in `cg/models/demultiplex/demux_results.py`
- `raw_index_dir`  in `cg/models/demultiplex/demux_results.py`
- `is_hiseq_x_copy_completed` in `cg/models/demultiplex/flow_cell.py`
- `is_hiseq_x_delivery_started` in `cg/models/demultiplex/flow_cell.py`
- `is_hiseq_x`  in `cg/models/demultiplex/flow_cell.py`
- `machine_number` in `cg/models/demultiplex/flow_cell.py`

I agree, lets remove them in subsequent PR:s. 
They can easily be found with
```
vulture --sort-by-size .  --exclude tests  > output.txt
```

**Comment by ChrOertlin** [comment]
Decision to make smaller PRs removing code, closing this one to avoid duplicated.

**Pull request closed without merging**